src/pages/on-chain-governance/index.page.tsx

```diff
@@ -11,8 +23,46 @@ import { OnChainGovernanceTitles } from "./enum/onChainGovernanceTitles";
 import { ProposalTable } from "./_components/ProposalTable";
 import { ProposalCards } from "./_components/ProposalCard";
 
-export default function OnChainGovernancePage({ votingCycle, proposals }) {
+interface OCGProps {
+  votingCycle: {
+    votingCycleNumber: number;
+    proposalsSubmitted: number;
+    dfips: number;
+    cfps: number;
+    currentStage: votingStages;
+    timeLeft: number;
+    totalTime: number;
+    totalVotes: number;
+  };
+  proposals: {
+    items: ProposalInfo[];
+    pages: CursorPage[];
+  };
+}
+export default function OnChainGovernancePage(props) {
+  const rpc = useWhaleRpcClient();
   const router = useRouter();
+  const [data, setData] = useState<OCGProps>({
+    votingCycle: props.votingCycle,
+    proposals: props.proposals,
+  });
+  const { votingCycle, proposals } = data;
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      void fetchData();
+    }, 15000);
```

Comment:
instead of arbitrary interval time, let's do something like
```
  const connection = useNetwork().connection;
  const interval = isPlayground(connection) ? 3000 : 30000;
```

Reply:
Using same logic as homepage fetch data call for listing blocks and txn

Reply:
removing auto refresh to reduce load on api